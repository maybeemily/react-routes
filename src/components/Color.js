import React from 'react';
import PropTypes from 'prop-types';

function Color({ match }) {
  const style = {
    backgroundColor: decodeURIComponent(match.params.color),
    width: '50vw',
    height: '50vw',
    fontSize: '4em',
    color: 'black'
  };

  return <div style={style}>{ style.backgroundColor }</div>;
}

Color.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      color: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Color;
