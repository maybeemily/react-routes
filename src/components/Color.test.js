import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('renders the color in the url', () => {
    const match = {
      params: {
        color: 'hotpink'
      }
    };
    const wrapper = shallow(<Color match={match}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
