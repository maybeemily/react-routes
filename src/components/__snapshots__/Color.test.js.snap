// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders the color in the url 1`] = `
<div
  style={
    Object {
      "backgroundColor": "hotpink",
      "color": "black",
      "fontSize": "4em",
      "height": "50vw",
      "letterSpacing": "1em",
      "textTransform": "uppercase",
      "width": "50vw",
    }
  }
>
  hotpink
</div>
`;
